src/ParseNode.js

```diff
@@ -290,13 +290,15 @@ type ParseNodeTypes = {
  * typing. Throws if the node's type does not match.
  */
 export function assertNodeType<NODETYPE: NodeType>(
-    node: ParseNode<*>,
+    // The union allows either ParseNode<*> or the union of two specific nodes.
+    node: ?ParseNode<*> | ParseNode<*>,
     type: NODETYPE,
 ): ParseNode<NODETYPE> {
     const typedNode = checkNodeType(node, type);
     if (!typedNode) {
         throw new Error(
-            `Expected node of type ${type}, but got node of type ${node.type}`);
+            `Expected node of type ${type}, but got ` +
+            (node ? `node of type ${node.type}` : String(node)));
```

Comment:
`String(node)` can be shortend to `${node}`.

Reply:
That's what I started with, but the flow type system disallows it.

Reply:
Note to self: `flow` doesn't allow implicit type coercion.